pkg/load/load.go

```diff
@@ -375,3 +349,15 @@ func chooseBestImageManifest(architecture string, index *ocispecs.Index) (ocispe
 
 	return ocispecs.Descriptor{}, errors.New("no manifests found")
 }
+
+// ReadyBy waits for at most `deadline` for an image registry to respond.
+// This is useful to check if the proxy container is ready to handle requests.
+func ReadyBy(ctx context.Context, addr string, deadline time.Duration) bool {
```

Comment:
This function is checking that the registry is ready by dialing `localhost`, however for environments with remote Docker daemons (CircleCI, Minikube), the CLI's localhost won't be the same as the proxy container.

I think instead we should probably make our pull call be able to retry some number of times, either on all errors or if the error is the one about not being able to connect.